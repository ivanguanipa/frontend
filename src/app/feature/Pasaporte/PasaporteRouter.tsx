import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { LazyFallback } from 'app/shared/components/LazyFallback';

const MainPage = React.lazy(() => import('./pages/Main'));
const ShowPage = React.lazy(() => import('./pages/Show'));

export const PasaporteRouter = () => (
  <React.Suspense fallback={<LazyFallback />}>
    {/* Layout compartido entre las rutas va aquÃ­ */}
    <Switch>
      <Route path="/pasaportes" exact component={MainPage}></Route>
      <Route path="/pasaportes/show/:id" component={ShowPage}></Route>

    </Switch>
  </React.Suspense>
);
