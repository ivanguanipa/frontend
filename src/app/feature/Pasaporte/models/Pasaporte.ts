type Nullable<T> = T | null ;
export interface Pasaporte {
    id?: Nullable<number>,
    fullname: string,
    address: string,
    birthdate: string,
    documentId: number,
    applicationDate: string,
    appointmentDate?: Nullable<string>,
    amount?: Nullable<number>,
}