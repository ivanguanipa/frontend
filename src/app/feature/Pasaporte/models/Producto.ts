export interface Producto {
  title: string;
  slug: string;
  body: string;
  createdAt: string;
  updatedAt: string;
}
