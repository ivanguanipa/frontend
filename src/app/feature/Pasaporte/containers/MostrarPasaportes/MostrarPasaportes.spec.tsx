import * as React from 'react';
import { ShallowWrapper, shallow, render, mount } from 'enzyme';
import { MostrarPasaportes } from './index';
import { Pasaporte } from 'app/feature/Pasaporte/models/Pasaporte';
import Router from "react-router-dom";
import { createMemoryHistory } from 'history';
import {PasaporteRepositorio} from 'app/core/api/pasaportes.repositorio';
import { axiosIntance } from 'app/core/config/AxiosConfig';
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: jest.fn().mockReturnValue({ id: 1}),
  })) 
  

describe('Mostrar Pasaporte Test', () => {
  let componentWrapper: ShallowWrapper;
  const pasaportes: Pasaporte = {
    fullname: 'lorem',
    address: 'ipsun',
    birthdate: '2021-01-01',
    document_id: 12345,
    application_date: '2021-01-01',
  };

   it('should match snapshot', done => {
    const mockCallback = jest.fn();
    const useEffectaa = jest.spyOn(React, "useEffect").mockImplementation(f => {
        f()
    });

    spyOn(axiosIntance, 'get').and.throwError('dsff')

    componentWrapper = shallow(
        <MostrarPasaportes 
        pasaporte={pasaportes} 
        mostrarPasaporte={mockCallback}
      />
      
    );
    expect(useEffectaa).toHaveBeenCalled();
    done()
  });
  
 
});

export {};
