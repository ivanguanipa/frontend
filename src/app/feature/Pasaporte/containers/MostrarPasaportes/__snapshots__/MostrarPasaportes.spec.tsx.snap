// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mostrar Pasaporte Test should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "action": "PUSH",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "2afqjr",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          Object {
            "hash": "",
            "key": "iy1nfk",
            "pathname": "/pasaportes/show/1",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 1,
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "iy1nfk",
          "pathname": "/pasaportes/show/1",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "key": "iy1nfk",
        "pathname": "/pasaportes/show/1",
        "search": "",
        "state": undefined,
      },
      "match": Object {
        "isExact": false,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <ContextProvider
    value={
      Object {
        "action": "PUSH",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "2afqjr",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          Object {
            "hash": "",
            "key": "iy1nfk",
            "pathname": "/pasaportes/show/1",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 1,
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "iy1nfk",
          "pathname": "/pasaportes/show/1",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <MostrarPasaportes
      mostrarPasaporte={[MockFunction]}
      pasaporte={
        Object {
          "address": "ipsun",
          "application_date": "2021-01-01",
          "birthdate": "2021-01-01",
          "document_id": 12345,
          "fullname": "lorem",
        }
      }
    />
  </ContextProvider>
</ContextProvider>
`;
